forem/forem
# Split ReCaptcha service and use verbs instead of pronouns

## What type of PR is this? (check all applicable)


## Description

Based on feedback received on #10502 ([see this conversation](https://github.com/forem/forem/pull/10502#discussion_r533615758)) this PR splits the `ReCaptcha` service in two.

I originally didn't notice [the service object docs](https://docs.forem.com/backend/service-objects/) that recommend we use verbs instead of nouns.

Also the approach here is a bit cleaner and less complex, i.e. it doesn't initialize an object and return `self` (this is useful sometimes so methods can be chained but not for this simple service object).

## QA Instructions, Screenshots, Recordings

- If the `SiteConfig` keys for email registration with reCAPTCHA are enabled you should run into the reCAPTCHA when trying to register.
- Try to report abuse (a comment on a Post) and if the reCAPTCHA SiteConfig keys are not configured you shouldn't see the reCAPTCHA appear in the form
   - Once keys are configured the reCAPTCHA should appear when logged out or the user has a vomit/too recent/banned
   - With keys configured the reCAPTCHA shouldn't appear when trying to report abuse as a Mod/Admin/"clean and non-recent user"

### UI accessibility concerns?

None. This only affects reCAPTCHA logic that currently exists

## Added tests?

- [x] No, and this is why: Existing tests cover this, they were just split in two (just like the service object)

## Added to documentation?


## [optional] What gif best describes this PR or how it makes you feel?

![same but different](https://media.giphy.com/media/C6JQPEUsZUyVq/giphy.gif)
